XamlControlsGallery/ControlExample.xaml.cs

```diff
@@ -260,6 +261,9 @@ private void FormatAndRenderSampleFromString(string sampleString, ContentPresent
             // Trim out stray blank lines at start and end.
             sampleString = sampleString.TrimStart('\n').TrimEnd();
 
+            // Also trim out spaces at the end of each line
+            sampleString = string.Join('\n', sampleString.Split('\n').Select(s => s.TrimEnd()));

```

Comment:
While I am not a fan to add even more processing for every sample, this is probably the best solution to fix this.

Reply:
Thanks for the review 👍

Yeah I agree that it'd be nice to minimize the amount of processing per sample, but this shouldn't really add much overhead anyway. Technically it's possible to optimize this with a single allocation (and also making it faster), by leveraging `ArrayPool<T>` and `Span<T>` APIs (specifically, `ReadOnlySpan<T>.IndexOf` and `ReadOnlySpan<T>.Trim`, though it seemed a bit overkill for this scenario and it would've required the additional `System.Memory` dependency, so I just left it at that for now 😄

Reply:
I think in the long term, instead of running the processing every time the sample is opened, we should have an automatic check that tells us what we need to change, so we can simply have the sample be "correct" and we don't have to run any correction steps on the users end.

Reply:
+1 to @chingucoding - the code change here is a great "catch-all", but even better would be if our samples were updated to be correct meaning this code never needs to actually be run. But for now, this is a reasonable change - thanks @sergio0694!

Reply:
Happy to help! 😊
I believe there should be a stylecop tool for that, the Windows Community Toolkit has that and it shows warnings whenever a line has any trailing whitespaces. You could probably just reuse that one, or at least use that as a starting point 👍

Reply:
Congrats @Sergio0694 on getting your first PR on the XCG merged!

Does stylecop also do that for x:String in XAML? Or does it do that for every file, regardless of type and content?

Reply:
Not a stylecop expert, but I believe it can be configured to work on specific file types.